### DISCUSSION TITLE
Swapping w/ nested field arrays leads to crash

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#8753] [General] [answered]
### Version Number

7.33.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-js-forked-mootdl?file=/src/App.js

### Steps to reproduce

Click the first swap button, it breaks the app. Second/third swap buttons work fine.

But if you comment out the default values, then it will work if you:
- Press add button twice
- Click the second make group button
- Click the first swap button

### Expected behaviour

Swap should work with nested fields with or without default values being set

### What browsers are you seeing the problem on?

Chrome

### Relevant log output
From my own app:
```
updateMounted
src/useController.ts:81
  78 |   const field: Field = get(control._fields, name);
  79 | 
  80 |   if (field) {
> 81 |     field._f.mount = value;
     | ^  82 |   }
  83 | };
  84 | 
  ```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
thanks for the issue report, please check if you have registered both inputs, which are `value` and `operator` inputs.  `useFieldArray` fields are not optional they should both be registered, this runtime error reveals that the input is not registered and mounted. we can patch or prevent this error, but this would potentially lead to other unexpected behavior.

**Reply by Moshyfawn** [reply]
I'm pretty sure it's something got to do with this _(can't check from mobile)_
```js
onChange={(e) => {
  field.onChange({
    ...field,
    value: e.target.value
  });
}}
```
Should be something along the lines of
```js
<input {...field} />

```

**Reply by fadi-george** [reply]
My example is simplified. For my company project, a lot of the components we have don't accept/forward refs so I can't use register.
Passing name to the fields doesn't seem to help.
https://codesandbox.io/s/group-2-mootdl?file=/src/ValueRow.js:641-647

**Reply by fadi-george** [reply]
Using a handler like this would work but ideally I could just swap if it didnt break
```
  const handleSwap = (indexA: number, indexB: number) => {
    const newGroup = [...group];
    const oldValue = newGroup[indexA];
    newGroup[indexA] = newGroup[indexB];
    newGroup[indexB] = oldValue;
    replace(newGroup);
  };
  ```

**Reply by fadi-george** [reply]
Spreading field ({...field}) also doesn't help.

**Reply by fadi-george** [reply]
Alt version w/ Controller: https://codesandbox.io/s/group-3-kxg6gw?file=/src/ValueRow.js

**Reply by bluebill1049** [reply]
In fact, a controlled component got registered without a mounted attribute, it's a sign of misusage. Can you provide both input defautlvalues and also make sure both inputs get registered?

**Reply by bluebill1049** [reply]
This a good example to be based on as well: https://codesandbox.io/s/react-hook-form-usefieldarray-ssugn

**Reply by fadi-george** [reply]
Ah right, I could split in to multiple useController dont need refs either: https://codesandbox.io/s/group-2-1-rloszt?file=/src/ValueRow.js
